src/executables/master_server_monitor.hpp

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * master_server_monitor.hpp -
+ */
+
+
+#include <cstring>
+#include <thread>
+#include <algorithm>
+#include <memory>
+#include <time.h>
+#include "heartbeat.h"
+#include "system_parameter.h"
+#include "connection_defs.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+class server_monitor
+{
+  public:
+
+    class server_entry;
+    class server_monitor_list;
+
+    static server_monitor &get_instance ()
+    {
+      static server_monitor instance;
+      return instance;
+    }
+
+    server_monitor (const server_monitor &) = delete;
+    server_monitor (server_monitor &&) = delete;
+
+    server_monitor &operator = (const server_monitor &) = delete;
+    server_monitor &operator = (server_monitor &&) = delete;
+
+    using server_entry_uptr_t = std::unique_ptr<server_monitor::server_entry>;
+
+    class server_entry
+    {
+      public:
+	server_entry (int pid, const char *server_name, const char *exec_path, char *args, CSS_CONN_ENTRY *conn);
+	~server_entry ();
+
+	server_entry (const server_entry &) = delete;
+	server_entry (server_entry &&) = delete;
+
+	server_entry &operator = (const server_entry &) = delete;
+	server_entry &operator = (server_entry &&) = delete;
+
+      private:
+	int m_pid;
+	std::string m_server_name;
+	std::string m_exec_path;
+	char **m_argv;
+	CSS_CONN_ENTRY *m_conn;
+	timeval m_last_revive_time;
+	bool m_need_revive;
```

Comment:
```suggestion
	const int m_pid;
	const std::string m_server_name;
	const std::string m_exec_path;
	char **m_argv;
	CSS_CONN_ENTRY *m_conn;
	const timeval m_last_revive_time;
	const bool m_need_revive;
```
Nit.
생성자에서 바로 대입되는 것들은 const로 하면 좋을 것 같습니다만... 연준님 의견에 따르겠습니다.

Reply:
`m_pid`, `m_server_name`, `m_exec_path`는 변경하였으나, `m_last_revive_time`이나 `m_need_revive`의 경우에는 추후 server 비정상 종료 시에 변경될 수 있는 변수이기 때문에 const로 설정할 수 없을 것 같습니다.